go/fixchain/logger.go

```diff
@@ -0,0 +1,232 @@
+package fixchain
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/google/certificate-transparency/go/x509"
+)
+
+// Limiter is an interface to allow different rate limiters to be used with the
+// Logger.
+type Limiter interface {
+	Wait()
+}
+
+// Logger contains methods to asynchronously log certificate chains to a
+// Certificate Transparency log and properties to store information about each
+// attempt that is made to post a certificate chain to said log.
+type Logger struct {
```

Comment:
It has...

gofmt doesn't format vertically mind you, it keeps the given field groupings and just aligns the types in each group.